### PR TITLE
refactor(peer_manager): move peer_store under peer_manager module

### PR DESCRIPTION
**Pull request by LNSD**
As part of the work done under #1293, the peer persistence database was decoupled from the message store database connection. This PR continues that work:

- [x] Moved `peer_store` module under `peer_manager` module. The code has been collocated with the peer manager code, as it is the only module depending on the peer store.
- [x] Decoupled `migrations.migrate()` procedure from the message store's.
- [x] Used a different migrations `SchemaVersion` than the message's store database. 
- [x] Updated the `wakunode2` module, the tests, and imports in general accordingly.

### CODE REVIEW

File: apps/wakunode2/wakunode2_setup_rest.nim [hunk 1]
```diff
@@ -13,7 +13,7 @@ import
 
 
 logScope:
-  topics = "wakunode.setup.rest"
+  topics = "wakunode rest"
```
Comment (jm-clius) [inline apps/wakunode2/wakunode2_setup_rest.nim:16]: I know not part of this PR, but FWIW I like that the app is just `wakunode` like here - I'm getting a bit wary of having a confusing `2` everywhere, when we're practically always working with version 2.

**Review by jm-clius — APPROVED** [review]

File: apps/wakunode2/wakunode2_setup_rest.nim (hunk 1 above)
Comment (LNSD) [reply apps/wakunode2/wakunode2_setup_rest.nim:16]: Yes... I also have that concern. To achieve that, we need to coordinate with the infra changes. I have avoided mixing too many infra changes for this release, we are already changing the configuration.

The related issue is: https://github.com/status-im/nwaku/issues/1002
